src/main/java/uk/gov/companieshouse/overseasentitiesapi/validation/HistoricalBeneficialOwnerValidator.java

```diff
@@ -0,0 +1,90 @@
+                    validateCeasedDate(historicalBeneficialOwnerDto.getCeasedDate(), errors, loggingContext);
+                    validateNotifiedDate(historicalBeneficialOwnerDto.getNotifiedDate(), errors, loggingContext);
+                    if (!historicalBeneficialOwnerDto.isCorporateIndicator()) {
+                        validateForename(historicalBeneficialOwnerDto.getForename(), errors, loggingContext);
+                        validateSurname(historicalBeneficialOwnerDto.getSurname(), errors, loggingContext);
+                    } else {
+                        validateCorporatename(historicalBeneficialOwnerDto.getCorporateName(), errors, loggingContext);
+                    }
+
+                }
+            }
+        }
+
+        return errors;
+    }
+
+    private boolean validateForename(String forename, Errors errors, String loggingContext) {
+        String qualifiedFieldName = getQualifiedFieldName(PARENT_FIELD, HistoricalBeneficialOwnerDto.FORENAME_FIELD);
+
+        return StringValidators.isNotBlank(forename, qualifiedFieldName, errors, loggingContext)
+                && StringValidators.isLessThanOrEqualToMaxLength(forename, 160, qualifiedFieldName, errors,
+                        loggingContext)
+                && StringValidators.isValidCharacters(forename, qualifiedFieldName, errors, loggingContext);
+    }
+
+    private boolean validateSurname(String surname, Errors errors, String loggingContext) {
+        String qualifiedFieldName = getQualifiedFieldName(PARENT_FIELD, HistoricalBeneficialOwnerDto.SURNAME_FIELD);
+
+        return StringValidators.isNotBlank(surname, qualifiedFieldName, errors, loggingContext)
+                && StringValidators.isLessThanOrEqualToMaxLength(surname, 160, qualifiedFieldName, errors,
+                        loggingContext)
+                && StringValidators.isValidCharacters(surname, qualifiedFieldName, errors, loggingContext);
+    }
+
+    private boolean validateCeasedDate(LocalDate ceasedDate, Errors errors, String loggingContext) {
+        String qualifiedFieldName = getQualifiedFieldName(PARENT_FIELD, HistoricalBeneficialOwnerDto.CEASED_DATE_FIELD);
+
+        return UtilsValidators.isNotNull(ceasedDate, qualifiedFieldName, errors, loggingContext)
+                && DateValidators.isDateInPast(ceasedDate, qualifiedFieldName, errors, loggingContext);
+    }
+
+    private boolean validateNotifiedDate(LocalDate notifiedDate, Errors errors, String loggingContext) {
+        String qualifiedFieldName = getQualifiedFieldName(PARENT_FIELD,
+                HistoricalBeneficialOwnerDto.NOTIFIED_DATE_FIELD);
+
+        return UtilsValidators.isNotNull(notifiedDate, qualifiedFieldName, errors, loggingContext)
+                && DateValidators.isDateInPast(notifiedDate, qualifiedFieldName, errors, loggingContext);
+    }
+
+    private boolean validateCorporatename(String corporatename, Errors errors, String loggingContext) {
```

Comment:
CorporateName should have capital N in method name and param

Reply:
thanks @markpit I have corrected it.